syntax = "proto3";

package ray.rpc;

message Event {
  
  enum SourceType {
    COMMON = 0;
    CORE_WORKER = 1;
    GCS = 2;
    RAYLET = 3;
  }

  enum Severity {
    INFO = 0;
    WARNING = 1;
    ERROR = 2;
    FATAL = 3;
  }
  
  // event_id is the unique ID of this event
  string event_id = 1;
  // job_id is the hex format string of JobID, which is generated by the GCS
  string job_id = 2;
  // node_id is the hex format string of ClientID, which is the remote client unique id
  string node_id = 3;
  // task_id is the hex format string of TaskID, which is a normal task or a driver task
  string task_id = 4; 
  
  // source type is the type of the source
  SourceType source_type = 5;
  // source_hostname is the hostname of the source
  string source_hostname = 6;
  // source_pid is the process pid of the source
  int32 source_pid = 7;

  // severity is the event severity level
  Severity severity = 8;
  // label describes some important information about this event, always use for the searching index and tag
  string label = 9;
  // message is main information of this event
  string message = 10;
  // timestamp is the report millionseconds since 00:00, Jan 1 1970 UTC
  int64 timestamp = 11;
  
}